src/app/map/shared/layer.service.ts

```diff
@@ -12,13 +15,63 @@ export class LayerService {
     osm: OSMLayer,
     vector: VectorLayer,
     xyz: XYZLayer,
+    wmts: WMTSLayer
   };
 
-  constructor() { }
+  getCapabilitiesStore: any[] = [];
+
+  constructor(private http: Http) { }
+
+  createLayer(options: LayerOptions): Observable<Layer> {
 
-  createLayer(options: LayerOptions): Layer {
     const layerCls = LayerService.layerClasses[options.type];
 
-    return new layerCls(options);
+    if (options.optionsFromCapabilities) {
+       return this.getCapabilities(options).map(
+        getCapabilities => {
+          return new layerCls(options, getCapabilities);
+        });
+    } else {
+      return new Observable(layer => layer.next(new layerCls(options)));
+    }
+  }
+
+  getCapabilities(options): Observable<any> {
+    let url = options.source.url + '?REQUEST=GetCapabilities&';
```

Comment:
You can create ad instance of URLSearchParams and pass it to the http.het method instead of appending strings to create the complete URL. See search/sources/search-source-nominatim.ts for an example.